pennylane/interfaces/torch.py

```diff
@@ -135,38 +170,47 @@ def forward(ctx, input_kwargs, *input_):
             return torch.from_numpy(res)
 
         @staticmethod
+        @once_differentiable
         def backward(ctx, grad_output):  # pragma: no cover
             """Implements the backwards pass QNode vector-Jacobian product"""
             # NOTE: This method is definitely tested by the `test_torch.py` test suite,
             # however does not show up in the coverage. This is likely due to
             # subtleties in the torch.autograd.FunctionMeta metaclass, specifically
             # the way in which the backward class is created on the fly
 
+            # determine the QNode variables which should be differentiated
+            diff_indices = None
+            non_diff_indices = set()
+
+            for arg, arg_variable in zip(ctx.saved_tensors, qnode.arg_vars):
+                if not getattr(arg, "requires_grad", False):
+                    indices = [i.idx for i in _flatten(arg_variable)]
+                    non_diff_indices.update(indices)
+
+            if non_diff_indices:
+                diff_indices = set(range(qnode.num_variables)) - non_diff_indices
+
             # evaluate the Jacobian matrix of the QNode
-            jacobian = qnode.jacobian(ctx.args, ctx.kwargs)
+            jacobian = qnode.jacobian(ctx.args, ctx.kwargs, wrt=diff_indices)
+            jacobian = torch.as_tensor(jacobian, dtype=grad_output.dtype)
 
-            if grad_output.is_cuda:  # pragma: no cover
-                grad_output_np = grad_output.cpu().detach().numpy()
-            else:
-                grad_output_np = grad_output.detach().numpy()
+            vjp = torch.transpose(grad_output.view(-1, 1), 0, 1) @ jacobian
+            vjp = vjp.flatten()
 
-            # perform the vector-Jacobian product
-            if not grad_output_np.shape:
-                temp = grad_output_np * jacobian
-            else:
-                temp = grad_output_np.T @ jacobian
+            if non_diff_indices:
+                # Torch requires we return a gradient of size (num_variables,)
+                res = torch.zeros([qnode.num_variables], dtype=grad_output.dtype)
```

Comment:
Are we going with zero rather than `None` for something that doesn't have a gradient evaluated?

Reply:
Yep! Previously, PyTorch didn't have much consistency. Some gradient functions would return `0`, some `None`, some would raise errors. Very recently, they decided to standardize it so that all unconnected gradients would be returned as tensors of zeros; see https://github.com/pytorch/pytorch/issues/30632#issuecomment-591652845.

Reply:
Oh nice, good to know! Thanks